dataplane/pkg/common/egress_interface.go

```diff
@@ -92,17 +96,20 @@ func parseProcFile(reader *bufio.Reader) (string, net.IP, error) {
 	return "", nil, errors.New("Failed to locate default route...")
 }
 
-func parseGatewayIP(defaultGateway string) net.IP {
+func parseGatewayIP(defaultGateway string) (net.IP, error) {
 	ip := net.IP{0, 0, 0, 0}
-	iv0, _ := strconv.ParseInt(defaultGateway[0:2], 16, 32)
-	iv1, _ := strconv.ParseInt(defaultGateway[2:4], 16, 32)
-	iv2, _ := strconv.ParseInt(defaultGateway[4:6], 16, 32)
-	iv3, _ := strconv.ParseInt(defaultGateway[6:], 16, 32)
-	ip[0] = byte(iv3)
-	ip[1] = byte(iv2)
-	ip[2] = byte(iv1)
-	ip[3] = byte(iv0)
-	return ip
+	if len(defaultGateway) == 8 {
+		iv0, _ := strconv.ParseInt(defaultGateway[0:2], 16, 32)
```

Comment:
I think we should get the errors from ParseInt here and check them. Otherwise we cannot guarantee if accurate value will be returned for ip. Some of the `iv` values may be `0` because of an error.